File: data_structures/java/stack.java
class node<obj>{
	node<obj> next;
	obj data;
	node(node<obj> n, obj a){
		next = n;
		data = a;
	}
	void setnext(node<obj> n){
		next = n;
	}
	void setele(obj a){
		data = a;
	}
	obj getele(){
		return data;
	}
	node<obj> getnext(){
		return next;
	}
}
class stack<obj>{
	private node<obj> top;
	private int size;
	stack(){
		top = new node<obj>(null, null);//sentinel node
		size=0;
	}
	void push(obj c){
		node<obj> n = new node<obj>(top, c);
		top=n;
		this.size++;
	}
	Object pop() throws NullPointerException{
		if(this.size == 0){
			throw new NullPointerException("Stack empty nothing to pop");
		}
		Object d = top.getele();
		top = top.getnext();
		this.size--;
		return d;
	}
	Object top() throws NullPointerException{
		if(this.size == 0){
			throw new NullPointerException("Stack empty");
		}
		return top.getele();
	}
	node<obj> getnext(){
		return top.next;
	}
	int size(){
		return this.size;
	}
}

File: data_structures/python/stack.py
class node:
	def __init__(self, Node, data):
		self.next = Node
		self.data = data
	def setnext(self, n):
		self.next = n
	def getele(self):
		return self.data
	def setele(self, data):
		self.data = data
	def getnext(self):
		return self.next

class stack:
    def __init__(s):#s denotes self
        s.top = node(None,None)#sentinal node, can be made without sentinal
        s.size = 0
    def push(s, d):
        n = node(s.top, d)
        s.top = n
        s.size += 1
    def pop(s):#removes top and returns data of it
        if(s.size == 0):
            raise Exception("Stack empty")
        else:
            d = s.top.data
            s.top = s.top.next
            s.size -=1
            return d
    def top(s):
        if(s.size == 0):
            raise Exception("Stack empty")
        else:
            return s.top.data
    def getnext(s):
        return s.top.next
